expandybird/expander/expander_test.go

```diff
@@ -50,23 +57,152 @@ func (etc *ExpanderTestCase) GetTemplate(t *testing.T) *common.Template {
 func GetOutputString(t *testing.T, description string) string {
 	output, err := ioutil.ReadFile(outputFileName)
 	if err != nil {
-		t.Errorf("cannot read output file for test case '%s': %s\n", description, err)
+		t.Fatalf("cannot read output file for test case '%s': %s", description, err)
 	}
 
 	return string(output)
 }
 
+func expandOutput(t *testing.T, actualOutput, description string) {
+	actualResult, err := NewExpansionResult(actualOutput)
+	if err != nil {
+		t.Fatalf("error in test case '%s': %s\n", description, err)
+	}
+
+	expectedOutput := GetOutputString(t, description)
+	expectedResult, err := NewExpansionResult(expectedOutput)
+	if err != nil {
+		t.Fatalf("error in test case '%s': %s\n", description, err)
+	}
+
+	if !reflect.DeepEqual(actualResult, expectedResult) {
+		message := fmt.Sprintf("want:\n%s\nhave:\n%s\n", expectedOutput, actualOutput)
+		t.Fatalf("error in test case '%s':\n%s\n", description, message)
+	}
+}
+
+func testExpandTemplateFromFile(t *testing.T, fileName, baseName string, importFileNames []string,
+	constructor func(string, io.Reader, []string) (*common.Template, error)) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		t.Fatalf("cannot open file %s: %s", fileName, err)
+	}
+
+	template, err := constructor(baseName, file, importFileNames)
+	if err != nil {
+		t.Fatalf("cannot create template from file %s: %s", fileName, err)
+	}
+
+	backend := NewExpander("../expansion/expansion.py")
```

Comment:
consider making this a top level variable like the others (validFileName, outputFileName, archiveFileName) so all the hardcoded things are defined in one place.